internal/controller/serving/reconcilers/llm_rolebinding_reconciler.go

```diff
@@ -0,0 +1,153 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/kmeta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/comparators"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/processors"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/resources"
+)
+
+var _ LLMSubResourceReconciler = (*LLMRoleBindingReconciler)(nil)
+
+type LLMRoleBindingReconciler struct {
+	LLMNoResourceRemoval
+	client             client.Client
+	roleBindingHandler resources.RoleBindingHandler
+	deltaProcessor     processors.DeltaProcessor
+}
+
+func NewLLMRoleBindingReconciler(client client.Client) *LLMRoleBindingReconciler {
+	return &LLMRoleBindingReconciler{
+		client:             client,
+		roleBindingHandler: resources.NewRoleBindingHandler(client),
+		deltaProcessor:     processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *LLMRoleBindingReconciler) Reconcile(ctx context.Context, log logr.Logger, llmisvc *kservev1alpha1.LLMInferenceService) error {
+	log.V(1).Info("Verifying that the model tier binding exists")
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(llmisvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, llmisvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *LLMRoleBindingReconciler) createDesiredResource(llmisvc *kservev1alpha1.LLMInferenceService) (*v1.RoleBinding, error) {
+	desiredRoleBinding := &v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kmeta.ChildName(llmisvc.Name, "-tier-binding"),
+			Namespace: llmisvc.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/managed-by": "odh-model-controller",
+			},
+		},
+		Subjects: []v1.Subject{
+			{
+				Kind:     "Group",
+				Name:     "system:serviceaccounts:openshift-ai-inference-tier-free",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+			{
+				Kind:     "Group",
+				Name:     "system:serviceaccounts:openshift-ai-inference-tier-premium",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+			{
+				Kind:     "Group",
+				Name:     "system:serviceaccounts:openshift-ai-inference-tier-enterprise",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		RoleRef: v1.RoleRef{
+			Kind:     "Role",
+			Name:     kmeta.ChildName(llmisvc.Name, "-model-user"),
```

Comment:
```suggestion
			Name:     kmeta.ChildName(llmisvc.Name, "-model-post-access"),
```

Reply:
Can we have one place where we construct the role name? That would reduce the risk of misconfiguration